osutil/disks/disks_linux.go

```diff
@@ -247,52 +247,66 @@ func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
 		if err != nil {
 			return nil, fmt.Errorf(errFmt, partMountPointSource, err)
 		}
+		dmUUID = bytes.TrimSpace(dmUUID)
 
 		dmName, err := ioutil.ReadFile(filepath.Join(dmDir, "name"))
 		if err != nil {
 			return nil, fmt.Errorf(errFmt, partMountPointSource, err)
 		}
+		dmName = bytes.TrimSpace(dmName)
+
+		// XXX: make the detection a helper/constant of kernel/fde
```

Comment:
it's definitely weird to need this code at this level, why should osutil/disks know about this?

Reply:
I'm not happy  about this myself. We do already know quite a bit of details about LUKS uuid details in this code but I will have a long hard look at this and maybe I can do something like change "IsDecryptedDevice" to a type that can be "none,luks,device-setup" and then at least it's not in the middle of all this other code. In any case this will need some serious refactor before it an go into master :/

Reply:
to be clear the knowledge about LUKS is less of an issue because that's a well known design. I understand that we have an asymmetry problem, but -device-unlock is something we made up.

Reply:
hmm, I wonder give the large comment I left in this function above about the fact that what we are doing here is kinda wrong and we should be parsing the format of the file depending on the "owner" if we could implement that code here for two possible owners, potentially extending this code to have a callback function or something for the different owners so that the code which knows about the format for -device-unlock owner is a dynamically registered callback defined in a related pkg, while the one for LUKS/cryptsetup we use here is a statically known one as part of the package osutil/disks package?

would be a fair amount of work though I fear

Reply:
I can look into this in a followup basically what we could do is:
1. verify that `/sys/dev/block/<major>:<minor>/dm/name` exists -> means the device is handled by device mapper
2. list directories in `/sys/dev/block/<major>:<minor>/slaves/*`, there should be just one, we expect a 1-1 mapping of device -> luks, or device -> linear mapper that device-unlock sets up (AFAIU it is possible that there is more 'slaves', but those are more complex setups)
3. read `/sys/dev/block/<major>:<minor>/slaves/<first-entry>/dev` to get the <major:minor> (aka parent-major-minor)
4. load udev properties of `/dev/block/<parent-major-minor>` and we get the disk which can be checked

In parallel to this I think we should call `dmsetup status /dev/block/<major>:<minor>` and inspect the output. In my system I see:
```
$ sudo dmsetup status /dev/block/254:0
0 485245263 crypt 
```
So we can either keep on looking at the name prefix, or inspect that the actual target == 'crypt'

I think my slight preference would be to do this in a followup though, also cleaning up the suffix check right below.